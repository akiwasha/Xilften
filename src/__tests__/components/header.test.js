import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import { Header } from '../../components';

jest.mock('react-router-dom');

describe('<Header />', () => {
  it('renders the basic <Header /> with a background', () => {
    const { container, getByText, getByTestId } = render(
      <Header>
        <Header.Frame>
          <Header.Logo src="/logo.svg" alt="Netflix" />
          <Header.TextLink active="true">Je suis un lien!</Header.TextLink>
        </Header.Frame>
      </Header>
    );

    expect(getByText('Je suis un lien!')).toBeTruthy();
    expect(getByTestId('header-bg')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders the basic <Header /> without a background', () => {
    const { container, getByText, queryByTestId } = render(
      <Header bg={false}>
        <Header.Frame>
          <Header.Logo src="/logo.svg" alt="Netflix" />
          <Header.ButtonLink>S'identifier</Header.ButtonLink>
          <Header.TextLink active="false">Je suis un lien!</Header.TextLink>
        </Header.Frame>
      </Header>
    );

    expect(getByText('Je suis un lien!')).toBeTruthy();
    expect(queryByTestId('header-bg')).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders the full <Header /> with a background', () => {
    const { container, getByText, getByTestId } = render(
      <Header src="joker1" dontShowOnSmallViewPort>
        <Header.Frame>
          <Header.Group>
            <Header.Logo src="/images/logo.svg" alt="xilften" />
            <Header.TextLink active={false} onClick={() => {}}>
              Series
            </Header.TextLink>
            <Header.TextLink active onClick={() => {}}>
              Films
            </Header.TextLink>
          </Header.Group>
          <Header.Group>
            <Header.Search searchTerm="Joker" setSearchTerm={() => {}} />
            <Header.Profile>
              <Header.Picture src="/images/plv.png" />
              <Header.Dropdown>
                <Header.Group>
                  <Header.Picture src="/images/plv.png" />
                  <Header.TextLink>Pierre-Laurent Vincent</Header.TextLink>
                </Header.Group>
                <Header.Group>
                  <Header.TextLink onClick={() => {}}>
                    Déconnection
                  </Header.TextLink>
                </Header.Group>
              </Header.Dropdown>
            </Header.Profile>
          </Header.Group>
        </Header.Frame>
        <Header.Feature>
          <Header.FeatureCallOut>
            Regardez Joker maintenant
          </Header.FeatureCallOut>
          <Header.Text>
            Dans les années 1980, à Gotham City, Arthur Fleck...
          </Header.Text>
          <Header.PlayButton>Lecture</Header.PlayButton>
        </Header.Feature>
      </Header>
    );

    expect(getByTestId('search-input')).toBeTruthy();
    expect(getByTestId('search-input').value).toBe('Joker');
    fireEvent.change(getByTestId('search-input'), {
      target: { value: ' Simpsons' },
    });
    fireEvent.click(getByTestId('search-click'));

    expect(getByText('Series')).toBeTruthy();
    expect(getByText('Films')).toBeTruthy();
    expect(getByText('Pierre-Laurent Vincent')).toBeTruthy();
    expect(getByText('Regardez Joker maintenant')).toBeTruthy();
    expect(
      getByText('Dans les années 1980, à Gotham City, Arthur Fleck...')
    ).toBeTruthy();
    expect(getByText('Lecture')).toBeTruthy();

    expect(container.firstChild).toMatchSnapshot();
  });
});
