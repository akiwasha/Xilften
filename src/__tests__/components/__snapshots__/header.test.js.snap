// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders the basic <Header /> with a background 1`] = `
<div
  class="header__Background-sc-17v3bsa-0 hkhSlY"
  data-testid="header-bg"
>
  <div
    class="header__Container-sc-17v3bsa-2 hVMkFs"
  >
    <p
      class="header__Link-sc-17v3bsa-4 bvVujV"
    >
      Je suis un lien!
    </p>
  </div>
</div>
`;

exports[`<Header /> renders the basic <Header /> without a background 1`] = `
<div
  class="header__Container-sc-17v3bsa-2 hVMkFs"
>
  <p
    class="header__Link-sc-17v3bsa-4 CZkiN"
  >
    Je suis un lien!
  </p>
</div>
`;

exports[`<Header /> renders the full <Header /> with a background 1`] = `
<div
  class="header__Background-sc-17v3bsa-0 keUSbR"
  data-testid="header-bg"
  src="joker1"
>
  <div
    class="header__Container-sc-17v3bsa-2 hVMkFs"
  >
    <div
      class="header__Group-sc-17v3bsa-1 hJSwMe"
    >
      <p
        class="header__Link-sc-17v3bsa-4 CZkiN"
      >
        Series
      </p>
      <p
        class="header__Link-sc-17v3bsa-4 CZkiN"
      >
        Films
      </p>
    </div>
    <div
      class="header__Group-sc-17v3bsa-1 hJSwMe"
    >
      <div
        class="header__Search-sc-17v3bsa-6 jKbBQv"
      >
        <button
          class="header__SearchIcon-sc-17v3bsa-7 fWTVno"
          data-testid="search-click"
        >
          <img
            alt="Search"
            src="/images/icons/search.png"
          />
        </button>
        <input
          class="header__SearchInput-sc-17v3bsa-9 cjBSUz"
          data-testid="search-input"
          placeholder="Rechercher des films et séries"
          value="Joker"
        />
      </div>
      <div
        class="header__Profile-sc-17v3bsa-10 bsgmJh"
      >
        <button
          class="header__Picture-sc-17v3bsa-3 gYBhWW"
          src="images/users//images/plv.png.png"
        />
        <div
          class="header__Dropdown-sc-17v3bsa-5 dkQdX"
        >
          <div
            class="header__Group-sc-17v3bsa-1 hJSwMe"
          >
            <button
              class="header__Picture-sc-17v3bsa-3 gYBhWW"
              src="images/users//images/plv.png.png"
            />
            <p
              class="header__Link-sc-17v3bsa-4 CZkiN"
            >
              Pierre-Laurent Vincent
            </p>
          </div>
          <div
            class="header__Group-sc-17v3bsa-1 hJSwMe"
          >
            <p
              class="header__Link-sc-17v3bsa-4 CZkiN"
            >
              Déconnection
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="header__Container-sc-17v3bsa-2 header__Feature-sc-17v3bsa-13 hVMkFs ipELMP"
  >
    <h2
      class="header__FeatureCallOut-sc-17v3bsa-15 fvkZyb"
    >
      Regardez Joker maintenant
    </h2>
    <p
      class="header__Text-sc-17v3bsa-14 kTeSZH"
    >
      Dans les années 1980, à Gotham City, Arthur Fleck...
    </p>
    <button
      class="header__PlayButton-sc-17v3bsa-8 fHGPQZ"
    >
      Lecture
    </button>
  </div>
</div>
`;
